x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/Match.java

```diff
@@ -63,6 +63,8 @@
 
 /**
  * Full text function that performs a {@link QueryStringQuery} .
+ * Unlike other ESQL functions, Match is both a function and an operator. This class provides implementation for both,
+ * but the operator documentation is generated in {@link MatchOperator}.
  */
 public class Match extends FullTextFunction implements PostOptimizationVerificationAware {
```

Comment:
I do wonder if Match should be abstract, with a MatchFunction and MatchOperator subclasses? Which would make the distinction between them more obvious.

Reply:
That's something we didn't do back in the day, mostly because there were fewer differences. I think it makes sense now. 

I implemented that separation in ed108863cad01d11199004ce37d3f1ec77d988ab. @fang-xing-esql @ChrisHegarty , do you think it makes sense?

Reply:
Makes sense to me.